src/main/java/com/github/invictum/reportportal/SuiteStorage.java

```diff
@@ -48,16 +49,54 @@ public void suiteFinisher(String id, Runnable finisher) {
      */
     public void finalizeActive() {
         suites.forEachKey(Runtime.getRuntime().availableProcessors(), id -> {
-            SuiteMetadata meta = suites.remove(id);
+            SuiteMetadata meta = suites.get(id);
+            if (meta.failedTests.isEmpty()) {
+                suites.remove(id);
+            }
```

Comment:
Just to clarify, this method removes only suite without failed tests, but always runs finisher even if suite has failed tests. Is it expected behaviour? Seems like finisher run should be inside if. Please correct me if I'm wrong.

Reply:
Yep. It's should work like code was written.